functions/util.py

```diff
@@ -7,18 +7,25 @@
 
 from .settings import SQLALCHEMY_DATABASE_URI
 
-_session = None
+_engine = None
 
 
-def get_db_session():
-    """Get the current SQLAlchemy session"""
-    global _session
+class sqlalchemy_session:
+    """Get a SQLAlchemy session from the connection pool"""
 
-    if not _session:
-        engine = create_engine(SQLALCHEMY_DATABASE_URI)
-        _session = sessionmaker(bind=engine)()
+    def __init__(self):
+        global _engine
 
-    return _session
+        if not _engine:
+            _engine = create_engine(SQLALCHEMY_DATABASE_URI)
+
+        self.session = sessionmaker(bind=_engine)()
+
+    def __enter__(self):
+        return self.session
+
+    def __exit__(self, type, value, traceback):
+        self.session.close()
```

Comment:
Cool! But shouldn't `__exit__` check for type of exception or traceback to rollback session or something?
Doesn't sqlalch provide something like this out of the box? 

And what about commit if there's no exception?

I think it might be expressed easier like
```
@contextmanager
def session_scope():
    """Provide a transactional scope around a series of operations."""
    session = Session()
    try:
        yield session
        session.commit()
    except:
        session.rollback()
        raise
    finally:
        session.close()
```
but your approach if fine, we just need to figure out commit and rollback for it, I guess.

Reply:
Ah, great points. Also, that syntax is cool -- didn't know about the `contextmanager` decorator. I'll make updates to this effect.